src/main/java/uk/ac/ebi/eva/evaseqcol/entities/SeqColExtendedDataEntity.java

```diff
@@ -144,6 +148,63 @@ public static SeqColExtendedDataEntity constructSeqColSequencesMd5Object(Assembl
         return seqColSequencesObject;
     }
 
+    /**
+     * Return the seqCol sorted-name-length-pairs extended object*/
+    public static SeqColExtendedDataEntity constructSeqColSortedNameLengthPairs(
+            SeqColExtendedDataEntity extendedNames, SeqColExtendedDataEntity extendedLengths) {
+        if (extendedNames.getExtendedSeqColData().getObject().size() != extendedLengths.getExtendedSeqColData().getObject().size()) {
+            return null; // Names and Lengths entities are not compatible
+        }
+        SeqColExtendedDataEntity SeqColSortedNameLengthPairsObject = new SeqColExtendedDataEntity().setAttributeType(
+                AttributeType.sortedNameLengthPairs);
+        JSONExtData seqColSortedNameLengthPairsArray = new JSONExtData();
+
+        // Get the plain name-length pairs
+        List<NameLengthPairEntity> nameLengthPairList = constructNameLengthPairList(extendedNames, extendedLengths);
+        // Get the sorted list
+        List<String> sortedNameLengthPairsList = constructSortedNameLengthPairs(nameLengthPairList);
+
+        SHA512ChecksumCalculator sha512ChecksumCalculator = new SHA512ChecksumCalculator();
+        seqColSortedNameLengthPairsArray.setObject(sortedNameLengthPairsList);
+        SeqColSortedNameLengthPairsObject.setExtendedSeqColData(seqColSortedNameLengthPairsArray);
+        SeqColSortedNameLengthPairsObject.setDigest(sha512ChecksumCalculator.calculateChecksum(
+                seqColSortedNameLengthPairsArray.toString()));
+        return SeqColSortedNameLengthPairsObject;
+    }
+
+    /**
+     * Retrieve and construct the list of name-length pairs*/
+    private static List<NameLengthPairEntity> constructNameLengthPairList(
+            SeqColExtendedDataEntity extendedNames, SeqColExtendedDataEntity extendedLengths) {
+        List<NameLengthPairEntity> nameLengthPairList = new ArrayList<>();
+        for (int i=0; i<extendedNames.getExtendedSeqColData().getObject().size(); i++) {
+            String name = extendedNames.getExtendedSeqColData().getObject().get(i);
+            String length = extendedLengths.getExtendedSeqColData().getObject().get(i);
+            nameLengthPairList.add(new NameLengthPairEntity(name, length));
+        }
+        return nameLengthPairList;
+    }
+
+    /**
+     * Return the sorted-name-length-pair list for the given list of nameLengthPairEntity*/
+    public static List<String> constructSortedNameLengthPairs(List<NameLengthPairEntity> nameLengthPairList) {
+        SHA512ChecksumCalculator sha512ChecksumCalculator = new SHA512ChecksumCalculator();
+        List<String> sortedNameLengthPairs = new ArrayList<>();
+        for (NameLengthPairEntity entity: nameLengthPairList) {
+            String nameLengthHash = sha512ChecksumCalculator.calculateChecksum(entity.toString());
```

Comment:
Should this use the `DigestCalculator` so that the string is JSON canonicalized? Then you can keep the `toString` method human-readable. It also seems to be what the [python implementation](https://github.com/refgenie/seqcol/blob/46675b669ae07db9da4fc3d113fefa2c1667b1fb/seqcol/seqcol.py#L300-L310) does.

Reply:
Absolutely, that's a good point.
I recently figured that out and created a **to-do** issue #32 to fix it.